thrust/system/cuda/detail/async/scan.h

```diff
@@ -0,0 +1,33 @@
+/******************************************************************************
```

Comment:
Do we really need this header? Can we get rid of it and just use granular includes?

Reply:
This matches the existing convention used for the sync scans and simplifies the ADL dispatch headers a bit. Let's consider the "separate algorithms per header" as an implementation detail of the CUDA backend here. I think it'd be nice to eventually split the headers out for both sync/async scans at some point as a cleanup.